packages/ra-ui-materialui/src/field/types.ts

```diff
@@ -25,7 +23,7 @@ export interface FieldProps<
      *     </List>
      * );
      */
-    sortBy?: Call<Objects.AllPaths, RecordType> | AnyString;
+    sortBy?: HintedString<ExtractRecordPaths<RecordType>>;
```

Comment:
This is a regression: we could order by anystring

Reply:
No `HintedString` does the same